adapter-base/src/main/java/org/eclipse/hono/adapter/AbstractProtocolAdapterBase.java

```diff
@@ -1031,8 +1077,8 @@ public void registerReadinessChecks(final HealthCheckHandler handler) {
         if (telemetrySender instanceof ServiceClient) {
             ((ServiceClient) telemetrySender).registerReadinessChecks(handler);
         }
-        if (eventSender instanceof ServiceClient) {
-            ((ServiceClient) eventSender).registerReadinessChecks(handler);
+        if (amqpEventSender instanceof ServiceClient) {
+            ((ServiceClient) amqpEventSender).registerReadinessChecks(handler);
```

Comment:
Nothing obvious as the connection is managed internally in the Kafka client.

Reply:
Can we use the Kafka client API to determine if the client is currently connected to the cluster?

Reply:
There is no way to ask the client that directly. There are a couple of workarounds I found like setting up a separate client that checks if it can connect to the cluster and access a topic, or checking the Prometheus metrics of the cluster. Quarkus implements such checks but disables them by default (https://quarkus.io/guides/kafka#kafka-health-check, https://github.com/smallrye/smallrye-reactive-messaging/blob/3518618b608717d07310fc2e58865ac19abef105/smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/health/BaseHealth.java#L34). 

I think the Kafka way is to try to produce records and if this fails to inspect the exception. We could think about failing the Kubernetes readiness check if producing repeatedly fails with the same connection error (consumers probably need different checks than producers). But I would like to collect more practical experience with our Kafka implementation before working on that topic. I could create an issue for it as the question exceeds the scope of this PR.

Reply:
I guess a TODO comment in the code should be enough ...